dashboard/app/models/school.rb

```diff
@@ -74,25 +74,150 @@ def self.get_seed_filename(stub_school_data)
     stub_school_data ? 'test/fixtures/schools.tsv' : 'config/schools.tsv'
   end
 
+  def self.construct_state_school_id(state_code, district_id, school_id)
+    "#{state_code}-#{district_id}-#{school_id}"
+  end
+
   # Seeds all the data from the source file.
   # @param options [Hash] Optional map of options.
   def self.seed_all(options = {})
     options[:stub_school_data] ||= CDO.stub_school_data
-    options[:force] ||= false
-
-    # use a much smaller dataset in environments that reseed data frequently.
-    schools_tsv = get_seed_filename(options[:stub_school_data])
-    expected_count = `wc -l #{schools_tsv}`.to_i - 1
-    raise "#{schools_tsv} contains no data" unless expected_count > 0
-
-    # It takes approximately 4 minutes to seed config/schools.tsv.
-    # Skip seeding if the data is already present. Note that this logic will
-    # not re-seed data if the number of records in the DB is greater than or
-    # equal to that in the TSV file, even if the data is different.
-    if options[:force] || School.count < expected_count
-      CDO.log.debug "seeding schools (#{expected_count} rows)"
+
+    if options[:stub_school_data]
+      # use a much smaller dataset in environments that reseed data frequently.
+      School.transaction do
+        merge_from_csv(get_seed_filename(true))
+      end
+    else
+      # NCES school data has been built up in the DB over time by pulling in different
+      # data files. This seeding recreates the order in which they we incorporated.
+      # NOTE: we are intentionally not populating the state_school_id based on the
+      # 2014-2015 preliminary or 2013-2014 public/charter data sets. Those files
+      # containt duplicate entries where some schools appear to be listed more than
+      # once but with different NCES ids. Since state_school_id needs to be unique
+      # the seeding would fail if we tried to set the state ids from those files.
+      # The 2014-2015 public/charter data does not have this issue so we do load the
+      # state_school_ids from there.
```

Comment:
Is this going to cause problems in the future? Do we have any follow up to do here?

Reply:
This could cause issues in the future if we add additional NCES data files and those files have duplicates (either within themselves or with schools we previously loaded.) If that happens then we will need a more involved fix. Likely that would mean some process to remove the duplicates from the data set.